src/Shared/UnitTests/TestEnvironment.cs

```diff
@@ -74,10 +74,6 @@ private void Cleanup()
                 foreach (var item in _invariants)
                     item.AssertInvariant(_output);
             }
-            else
-            {
-                throw new InvalidOperationException("Already disposed");
```

Comment:
Was this failing anywhere? Why?

Reply:
Yes, when an invariant failed, then the exception would stop `GC.SuppressFinalize` from being called, and so `Cleanup` would be called again from the destructor.